### PR TITLE
New initial assumption

### LINKED ISSUE #1341: enh: Design effective initial assumption
The initial assumtions should be actualized using most actual exemples from other AutoML (moslty boosting-based).

The effect should be confirmed on AutoMLBenchmark.

### PR DESCRIPTION
**Pull request by dmitryglhf**
This is a 🔨 code refactoring.

## Summary

This PR introduces the following key updates:

- **New Initial Assumptions:** Updates initial assumptions by adding boosting-based solutions.
  
  Comparison table between old and new assumptions (validated on automlbenchmark):
  |    | Metric (mean)            |   main |   gbm |
  |---:|:------------------|-------------:|-------------------:|
  |  0 | auc               |     0.869263 |           0.879746 |
  |  1 | acc               |     0.84667  |           0.852339 |
  |  2 | balacc            |     0.805336 |           0.822745 |
  |  3 | logloss           |     0.449189 |           0.377827 |
  |  4 | training_duration |   242.554    |        251.445     |

## Context

  Closes #1341

### CODE REVIEW

File: fedot/api/api_utils/assumptions/task_assumptions.py [hunk 1]
```diff
@@ -90,11 +90,64 @@ class RegressionAssumptions(TaskAssumptions):
...
+
+        # Constants
+        CATBOOSTREG = 'catboostreg'
+        XGBOOSTREG = 'xgboostreg'
+        LGBMREG = 'lgbmreg'
+        RFR = 'rfr'
+        RIDGE = 'ridge'
+
+        # Parameters of models
+        models_params = {
```
Comment (nicl-nno) [inline fedot/api/api_utils/assumptions/task_assumptions.py:104]: Это какие-то эффективные гиперпараметры?

File: fedot/api/api_utils/assumptions/task_assumptions.py [hunk 2]
```diff
@@ -90,11 +90,64 @@ class RegressionAssumptions(TaskAssumptions):
...
+            },
+            RIDGE: {}
         }
 
+        # Get composite assumptions
+        assumptions['gbm'] = PipelineBuilder() \
+            .add_branch((CATBOOSTREG, models_params[CATBOOSTREG]),
+                        (XGBOOSTREG, models_params[XGBOOSTREG]),
+                        (LGBMREG, models_params[LGBMREG])) \
+            .join_branches(CATBOOSTREG, models_params[CATBOOSTREG])
```
Comment (nicl-nno) [inline fedot/api/api_utils/assumptions/task_assumptions.py:141]: А почему модели обьединены CATBOOSTREG, а не линейной моделью?

**Comment by nicl-nno** [comment]
image

А какая модель тут была раньше, что качество так просело?

File: fedot/api/api_utils/assumptions/task_assumptions.py (hunk 1 above)
Comment (dmitryglhf) [reply fedot/api/api_utils/assumptions/task_assumptions.py:104]: При тестировании пайплайнов на Kaggle с указанными гиперпараметрами значение метрики немного улучшалось. Настройку параметров прекращал, когда скорость работы значительно увеличивалась или метрика падала. Для CatBoost и линейных моделей параметры остались по умолчанию, поскольку они изначально показывали хорошие результаты, а при подборе метрика ухудшалась и скорость подбора возрастала (для CatBoost).

File: fedot/api/api_utils/assumptions/task_assumptions.py (hunk 2 above)
Comment (dmitryglhf) [reply fedot/api/api_utils/assumptions/task_assumptions.py:141]: Пробовал объединять с помощью Random Forest и линейных моделей. В обоих случаях метрика немного ухудшалась. Аналогичный результат был, когда эти модели были перед разветвлением. Это тестирование проводилось не на полном бенчмарке, а на Kaggle. Думаю здесь мне стоит провести больше тестов с объединением линейной моделью.

**Comment by dmitryglhf** [comment]
> image
> 
> А какая модель тут была раньше, что качество так просело?

Раньше здесь было Scaling + RandomForest

**Comment by nicl-nno** [comment]
> Раньше здесь было Scaling + RandomForest.

Вроде он и сейчас в начальной популяции есть. Он не вообще используется или просто проигрывает в ходе отбора?

File: fedot/api/api_utils/assumptions/task_assumptions.py (hunk 2 above)
Comment (nicl-nno) [reply fedot/api/api_utils/assumptions/task_assumptions.py:141]: Ну вот в презентации расхваливалась именно линейная модель. Можно конечно оба варианта добавлять в популяцию, с использованием кэша это вычислительно не очень дорого.

Comment (dmitryglhf) [reply fedot/api/api_utils/assumptions/task_assumptions.py:141]: Хорошо, я протестирую это.

**Comment by dmitryglhf** [comment]
> > Раньше здесь было Scaling + RandomForest.
> 
> Вроде он и сейчас в начальной популяции есть. Он не вообще используется или просто проигрывает в ходе отбора?

Сейчас он есть, но не использовался, в бенчмарке тестировалось только 'gbm' приближение.

**Comment by dmitryglhf** [comment]
|    | Metric            |       main |   gbm_linear |   linear_gbm_linear |   linear_gbm_catboost |
|---:|:------------------|-----------:|-------------:|--------------------:|----------------------:|
|  0 | auc               |   0.869263 |     0.879204 |            0.853338 |              0.848935 |
|  1 | acc               |   0.84667  |     0.851591 |            0.826598 |              0.821353 |
|  2 | balacc            |   0.805336 |     0.823641 |            0.79243  |              0.793106 |
|  3 | logloss           |   0.449189 |     0.381866 |            0.504397 |              0.492602 |
|  4 | training_duration | 242.554    |  1007.77     |           86.6531   |             99.8162   |

Pipelines (ridge instead of logit in regression tasks);
__'main' assumption:__
image

__'gbm_linear' assumption:__
image

__'linear_gbm_linear' assumption:__
image

__'linear_gbm_catboost' assumption:__
image

<details>

<summary>Full table and Details</summary>

[main_vs_gl.csv](https://github.com/user-attachments/files/18092810/main_vs_gl.csv)
[main_vs_lgl.csv](https://github.com/user-attachments/files/18092869/main_vs_lgl.csv)
[main_vs_lgc.csv](https://github.com/user-attachments/files/18092868/main_vs_lgc.csv)

</details>


Chose `gbm_linear` as result.

**Comment by dmitryglhf** [comment]
> Metric	main	gbm_linear	linear_gbm_linear	linear_gbm_catboost
> 0	auc	0.869263	0.879204	0.853338	0.848935
> 1	acc	0.84667	0.851591	0.826598	0.821353
> 2	balacc	0.805336	0.823641	0.79243	0.793106
> 3	logloss	0.449189	0.381866	0.504397	0.492602
> 4	training_duration	242.554	1007.77	86.6531	99.8162
> Pipelines (ridge instead of logit in regression tasks); **'main' assumption:** image
> 
> **'gbm_linear' assumption:** image
> 
> **'linear_gbm_linear' assumption:** image
> 
> **'linear_gbm_catboost' assumption:** image
> 
> Full table and Details
> Chose `gbm_linear` as result.

Nodes fit time in `gbm_linear` for [Kaggle_s4e6](https://www.kaggle.com/competitions/playground-series-s4e6/overview)

Node: scaling, fit time: 1.89 seconds
Node: catboost, fit time: 82.914 seconds
Node: lgbm, fit time: 7.416 seconds
Node: xgboost, fit time: 2.812 seconds
Node: logit, fit time: 0.551 seconds
2024-12-11 18:14:40,666 - ApiComposer - Initial pipeline was fitted in 106.3 sec.

**Comment by dmitryglhf** [comment]
|    | Metric            |   main_scal_rf |   gbm_linear |   gbm_catboost_new_params |   rf3_gbm_linear |   xgb_lgbm_linear |
|---:|:------------------|---------------:|-------------:|--------------------------:|-----------------:|------------------:|
|  0 | auc               |       0.869263 |     0.879204 |                  0.879746 |         0.86393  |          0.877597 |
|  1 | acc               |       0.84667  |     0.851591 |                  0.852339 |         0.837973 |          0.848727 |
|  2 | balacc            |       0.805336 |     0.823641 |                  0.822745 |         0.805042 |          0.818386 |
|  3 | logloss           |       0.449189 |     0.381866 |                  0.377827 |         0.625611 |          0.392734 |
|  4 | training_duration |     242.554    |  1007.77     |                251.445    |       104.559    |        213.057    |

<details>

<summary>Full table and Details</summary>

[catboost_new_params.csv](https://github.com/user-attachments/files/18238676/catboost_new_params.csv)
[rf3_gbm_linear.csv](https://github.com/user-attachments/files/18238677/rf3_gbm_linear.csv)
[xgb_lgbm_linear.csv](https://github.com/user-attachments/files/18238678/xgb_lgbm_linear.csv)


`rf3_gbm_linear`:
image

`xgb_lgbm_linear`:
image

</details>
